examples/material_gallery/lib/demo/markdown_demo.dart

```diff
@@ -0,0 +1,46 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_markdown/flutter_markdown.dart';
+
+class MarkdownDemo extends StatefulComponent {
+  MarkdownDemo({Key key}) : super(key: key);
+
+  _MarkdownDemoState createState() => new _MarkdownDemoState();
+}
+
+class _MarkdownDemoState extends State<MarkdownDemo> {
+
+  void initState() {
+    super.initState();
+
+    DefaultAssetBundle.of(context).loadString('packages/flutter_gallery_assets/markdown.md').then((String fileContents) {
+      setState(() {
```

Comment:
Yes, the closure will hold a reference to the object and prevent it from being deleted.  However, that doesn't control whether the parent decides to remove the widget from the tree.  If the parent removes the widget from the tree, it will no longer be mounted and calling setState will trigger an assert.